cmake/alpakaCommon.cmake

```diff
@@ -14,6 +14,49 @@ include(CMakePrintHelpers) # for easier printing of variables and properties
 #-------------------------------------------------------------------------------
 # Options.
 
+# Compiler options
+function(alpaka_compiler_option name description default)
+    if(NOT DEFINED name)
+        set(ALPAKA_${name} ${default} CACHE STRING "${description}")
+        set_property(CACHE ALPAKA_${name} PROPERTY STRINGS "DEFAULT;ON;OFF")
+    endif()
+endfunction()
+
+# Add parameter to a variable or target
+#
+# This method is automatically documenting all parameter added to the compiler into the variables
+# ALPAKA_COMPILER_OPTIONS_HOST, ALPAKA_COMPILER_OPTIONS_DEVICE, and ALPAKA_COMPILER_OPTIONS_HOST_DEVICE.
+#
+# scope - which compiler is effected: DEVICE, HOST, or HOST_DEVICE
+# operation - operation used to append given parameter: set, list, or target
+# name - name of the variable or target
+# ... - parameter to appended to the variable or target 'name'
+function(alpaka_set_compiler_options scope operation name)
+    if(scope STREQUAL HOST)
+        set(ALPAKA_COMPILER_OPTIONS_HOST ${ALPAKA_COMPILER_OPTIONS_HOST} ${ARGN} PARENT_SCOPE)
```

Comment:
I prefer `list(APPEND ...)`, but I guess that does not work if the list does not exist and cannot inject into the parent scope.

Reply:
`set` is equal but required because I need to move the result into the parent scope.